tests.py

```diff
@@ -114,6 +114,75 @@ def setUpClass(self):
         self.search_ranges = {key: [getattr(self, key)] for key in self.search_keys}
 
 
+class TestInjectionParametersGenerator(BaseForTestsWithOutdir):
+    label = "TestInjectionParametersGenerator"
+    class_to_test = pyfstat.InjectionParametersGenerator
+
+    def test_numpy_priors(self):
+        numpy_priors = {
+            "ParameterA": {"uniform": {"low": 0.0, "high": 0.0}},
+            "ParameterB": {"uniform": {"low": 1.0, "high": 1.0}},
+        }
+        self.InjectionGenerator = self.class_to_test(numpy_priors)
+
+        parameters = self.InjectionGenerator.draw()
+        print(parameters)
+        self.assertTrue(parameters["ParameterA"] == 0.0)
+        self.assertTrue(parameters["ParameterB"] == 1.0)
+
+    def test_callable_priors(self):
+        callable_priors = {"ParameterA": lambda: 0.0, "ParameterB": lambda: 1.0}
+        self.InjectionGenerator = self.class_to_test(callable_priors)
+
+        parameters = self.InjectionGenerator.draw()
+        self.assertTrue(parameters["ParameterA"] == 0.0)
+        self.assertTrue(parameters["ParameterB"] == 1.0)
+
+    def test_constant_priors(self):
+        constant_priors = {"ParameterA": 0.0, "ParameterB": 1.0}
+        self.InjectionGenerator = self.class_to_test(constant_priors)
+
+        parameters = self.InjectionGenerator.draw()
+        self.assertTrue(parameters["ParameterA"] == 0.0)
+        self.assertTrue(parameters["ParameterB"] == 1.0)
+
+    def test_rng_seed(self):
+        a_seed = 420
+
+        samples = []
+        for i in range(2):
+            self.InjectionGenerator = self.class_to_test(
+                priors={"ParameterA": {"normal": {"loc": 0, "scale": 1}}}, seed=a_seed
+            )
+            samples.append(self.InjectionGenerator.draw())
+        self.assertTrue(samples[0]["ParameterA"] == samples[1]["ParameterA"])
```

Comment:
When called from `TestAllSkyInjectionParametersGenerator`, this still only checks `ParameterA`, which is actually ok, but Alpha and Delta ignore the seed:
```
>>> from pyfstat import AllSkyInjectionParametersGenerator
>>> asipg=AllSkyInjectionParametersGenerator(priors={"ParameterA": {"normal": {"loc": 0, "scale": 1}}}, seed=42)
22:06 INFO    : Updated parameters: Alpha Delta
22:06 INFO    : Updated parameters: ParameterA
>>> asipg.draw()
{'Alpha': 1.8296262331751116, 'Delta': -0.7278825693693345, 'ParameterA': 0.30471707975443135}
>>> asipg=AllSkyInjectionParametersGenerator(priors={"ParameterA": {"normal": {"loc": 0, "scale": 1}}}, seed=42)
22:06 INFO    : Updated parameters: Alpha Delta
22:06 INFO    : Updated parameters: ParameterA
>>> asipg.draw()
{'Alpha': 5.337303292736909, 'Delta': -1.3868702724508961, 'ParameterA': 0.30471707975443135}
```